### PR TITLE
Feature/tao 8945/online balancer

### PR DESCRIPTION
**Pull request by bziondik**
Related to: https://oat-sa.atlassian.net/browse/TAO-8945

For `online` mode was added a balancer for getting API URLs.
Was added a logical load balancer, based on the user login - `characterBalancer.js `
For compatibility with existing modes was added `defaultBalancer.js` 
See configuration example in `config/config.online.balancer.sample.json `

### CODE REVIEW

File: src/core/balancerProviders/characterBalancer.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                return new TypeError('characterBalancer should have a positive or 0 numeric position setting');
+            }
+            if(typeof login !== 'string' || login.length <= 0) {
+                console.error('invalid login');
+                return new TypeError('invalid login');
+            }
+            const key = login.charAt(this.getConfig().balancer.position);
+            const index = this.getConfig().balancer.map[key];
+            if(index !== undefined) {
+                return this.getConfig().api[index];
```
Comment (jsconan) [inline src/core/balancerProviders/characterBalancer.js]: Maybe should you also check if the api contains the expected index.

File: src/core/balancerProviders/defaultBalancer.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+         *  "lti": "..."
+         * }
+        */
+        getApi() {
+            return this.getConfig().api;
+        }
+    };
+};
+
+export default defaultrBalancer;
```
Comment (jsconan) [inline src/core/balancerProviders/defaultBalancer.js]: small typo in the name

File: src/services/login/adapter/online.js [hunk 3]
```diff
@@ -19,8 +19,17 @@
...
+import balancerFactory  from '../../../core/balancerFactory';
 
 export default function onlineLoginAdapterFactory() {
+    let api;
+    const errorPromise = (error) => {
+        const err = error || new Error('Problem with balancer API');
+        err.status = 401;
+        return new Promise( (resolve, reject) => {
+            return reject(err);
+        });
```
Comment (jsconan) [inline src/services/login/adapter/online.js]: No need to use such a complex construction. You can simply `return Promise.reject(err);`

File: src/core/balancerProviders/defaultBalancer.js (hunk 2 above)
Comment (bziondik) [reply src/core/balancerProviders/defaultBalancer.js]: fixed

File: src/services/login/adapter/online.js (hunk 3 above)
Comment (bziondik) [reply src/services/login/adapter/online.js]: fixed

**Pull request merged**